src/player/Player.php

```diff
@@ -969,6 +979,27 @@ public function doChunkRequests() : void{
 		}
 	}
 
+	public function getDeathPosition() : ?Position{
+		return $this->deathPosition;
+	}
+
+	/**
+	 * @param Vector3|Position|null $pos
+	 */
+	public function setDeathPosition(?Vector3 $pos) : void{
+		if($pos !== null){
+			if($pos instanceof Position && $pos->world !== null){
```

Comment:
I think it's fine to just assert that the position must be valid here. We shouldn't be setting death position to an unloaded world anyway.

Reply:
Apparently this is the same as how the spawn setting logic works. Perhaps we need to take a hard look at why Position is allowed to have a null `World`...